src/walletSdk/Horizon/Transaction/SponsoringBuilder.ts

```diff
@@ -0,0 +1,74 @@
+import StellarSdk, {
+  TransactionBuilder as StellarTransactionBuilder,
+  Transaction,
+} from "stellar-sdk";
+import { IssuedAssetId } from "../../Asset";
+
+import { CommonTransactionBuilder } from "./CommonTransactionBuilder";
+import { AccountKeypair } from "../Account";
+
+export class SponsoringBuilder extends CommonTransactionBuilder<SponsoringBuilder> {
+  private sponsorAccount: AccountKeypair;
+  private builder: StellarTransactionBuilder;
+
+  constructor(
+    sponsoredAddress: string,
+    sponsorAccount: AccountKeypair,
+    builder: StellarTransactionBuilder,
+  ) {
+    super(sponsoredAddress);
+    this.sponsorAccount = sponsorAccount;
+    this.builder = builder;
+
+    this.startSponsoring();
+  }
+
+  createAccount(
+    newAccount: AccountKeypair,
+    startingBalance: number = 0,
+  ): SponsoringBuilder {
+    this.operations.push(
+      StellarSdk.Operation.createAccount({
+        destination: newAccount.publicKey,
+        startingBalance: startingBalance.toString(),
+        source: this.sponsorAccount.publicKey,
+      }),
+    );
+    return this;
+  }
+
+  addAssetSupport(asset: IssuedAssetId, trustLimit?: string) {
```

Comment:
Is it necessary to call super here? Can't we do `return this` in the parent class?

Reply:
I tried this a bunch last night and for some reason Typescript didn't like it 🤦‍♂️. But yup definitely cleaner this way:
https://github.com/stellar/typescript-wallet-sdk/pull/45/commits/503e00e70e25220539d2e3d7e96a2cee13f7643d